tox.ini

```diff
@@ -4,12 +4,21 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py37,py38,py39,py310
+envlist = 
+    py37
+    py38
+    py39
+    py310
+isolated_build = True
 
 [testenv]
-deps =
-    pytest
-    coverage
-    typing_extensions
+extras =
+    tox_only
```

Comment:
I think this crystallizes part of why i don't like this approach.

Reading this, i don't know from `tox`'s own configuration file what `tox` needs to run. I have to go find out what dependencies `tox` will need from another file, a file which is not typically used by `tox`. Maybe this part is just growing pains for a new approach, but reading this, i didn't actually know that i would need to look at the `tools.poetry.extras` section specifically for `tox_only`. 

Without this change, i like that `tox` tells me everything i need to know about it within its `tox.ini` file, and same with `pre-commit`. I don't like the amount of coupling across seemingly unrelated infrastructure that is introduced with this change.

Reply:
>  I don't know from tox's own configuration file what tox needs to run

This tells tox to install the package based on what is configured in the pyproject.toml.  It's essentially letting the screenpy package dictate what deps get installed rather than having to be configured separately.  This is the beginning of a single source of truth. 

At some point the whole tox configuration sections will reside in the pyproject.toml as well which will make things a little less awkward to look at.

Reply:
I really should  label that extras group as 'test'.